framework/src/modules/nft/method.ts

```diff
@@ -295,4 +296,128 @@ export class NFTMethod extends BaseMethod {
 			collectionID,
 		});
 	}
+
+	public async lock(methodContext: MethodContext, module: string, nftID: Buffer): Promise<void> {
+		const nftStore = this.stores.get(NFTStore);
+
+		const nftExists = await nftStore.has(methodContext, nftID);
+
+		if (!nftExists) {
+			this.events.get(LockEvent).error(
+				methodContext,
+				{
+					module,
+					nftID,
+				},
+				NftEventResult.RESULT_NFT_DOES_NOT_EXIST,
+			);
+
+			throw new Error('NFT substore entry does not exist');
+		}
+
+		const owner = await this.getNFTOwner(methodContext, nftID);
+
+		if (owner.length === LENGTH_CHAIN_ID) {
+			this.events.get(LockEvent).error(
+				methodContext,
+				{
+					module,
+					nftID,
+				},
+				NftEventResult.RESULT_NFT_ESCROWED,
+			);
+
+			throw new Error('NFT is escrowed to another chain');
+		}
+
+		const userStore = this.stores.get(UserStore);
+		const userKey = userStore.getKey(owner, nftID);
+		const userData = await userStore.get(methodContext, userKey);
+
+		if (userData.lockingModule !== NFT_NOT_LOCKED) {
+			this.events.get(LockEvent).error(
+				methodContext,
+				{
+					module,
+					nftID,
+				},
+				NftEventResult.RESULT_NFT_LOCKED,
+			);
+
+			throw new Error('NFT is already locked');
+		}
+
+		userData.lockingModule = module;
+
+		await userStore.set(methodContext, userKey, userData);
+
+		this.events.get(LockEvent).log(methodContext, {
+			module,
+			nftID,
+		});
+	}
+
+	public async unlock(methodContext: MethodContext, module: string, nftID: Buffer): Promise<void> {
+		const nftStore = this.stores.get(NFTStore);
+
+		const nftExists = await nftStore.has(methodContext, nftID);
+
+		if (!nftExists) {
+			this.events.get(LockEvent).error(
+				methodContext,
+				{
+					module,
+					nftID,
+				},
+				NftEventResult.RESULT_NFT_DOES_NOT_EXIST,
+			);
+
+			throw new Error('NFT substore entry does not exist');
+		}
+
+		const nftData = await nftStore.get(methodContext, nftID);
+
+		if (nftData.owner.length === LENGTH_CHAIN_ID) {
+			throw new Error('NFT is escrowed to another chain');
+		}
+
+		const userStore = this.stores.get(UserStore);
+		const userKey = userStore.getKey(nftData.owner, nftID);
+		const userData = await userStore.get(methodContext, userKey);
```

Comment:
I was wondering why you redefined this logic instead of calling `getLockingModule` which does the same thing and assumed it was because you want to throw a different error message as previously suggested. But since the error message is not different what is the reason/benefit for redefining this logic ?

Reply:
I avoided the use of `getLockingModule` because it has redundant logic to verify if NFT exists which is already performed separately to log events.

Reply:
true